pennylane/circuit_graph.py

```diff
@@ -137,6 +137,34 @@ def __init__(self, ops, variable_deps):
                 # Create an edge between this and the previous operator
                 self._graph.add_edge(wire[i - 1], wire[i])
 
+    def print_contents(self):
```

Comment:
The one might even be moved to `utils.py` and made a function that takes `ops` and `observables`. This way one could use it without having to construct a `CircuitGraph` and it could also be separated from this class as well.

Reply:
I was thinking to move most of the printing code to `Operation.__repr__`, so the ops know how to print themselves.

Reply:
I like that idea! It's always a lot of effort when you need to print out operations.

Reply:
Sounds good!